master/internal/resourcemanagers/agent/agent.go

```diff
@@ -92,16 +95,31 @@ func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 		socket, ok := msg.Accept(ctx, aproto.MasterMessage{}, true)
 		check.Panic(check.True(ok, "failed to accept websocket connection"))
 		a.socket = socket
+
 		lastColonIndex := strings.LastIndex(msg.Ctx.Request().RemoteAddr, ":")
 		if lastColonIndex == -1 {
 			a.address = msg.Ctx.Request().RemoteAddr
 		} else {
 			a.address = msg.Ctx.Request().RemoteAddr[0:lastColonIndex]
 		}
-		wsm := ws.WriteMessage{Message: aproto.AgentMessage{MasterSetAgentOptions: a.opts}}
+
+		var masterSetAgentOptions aproto.AgentMessage
+		if a.awaitingReconnect && a.agentReattachEnabled {
```

Comment:
trying to read this carefully and think - previously if an agent had an intermittent network failure, we would not lose the containers (due to the reconnect/replay pr i landed). now if one has a network failure with a.agentReattachEnabled == false, i think when it reconnects we will drop all containers? we only need to clear all when the agent has forgotten its containers, right?

Reply:
@ioga this is the only comment that is critical i think